From 42c9f6a5e50b1b1066e6f67a3da26e3ab6d79aff Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:23:29 -0400
Subject: [tools] perf/build: split out feature check 'fortify-source'

Message-id: <1408462094-14194-41-git-send-email-jolsa@redhat.com>
Patchwork-id: 88030
O-Subject: [PATCH RHEL7.1 BZ1131394 040/325] tools/perf/build: Split out feature check: 'fortify-source'
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 1ea6f99efd8ae61fce68c97a9cf9f722cfbca3ad
Author: Ingo Molnar <mingo@kernel.org>
Date: Mon Oct 7 09:38:28 2013 +0200
---
 create mode 100644 tools/perf/config/feature-checks/test-fortify-source.c

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/config/Makefile b/tools/perf/config/Makefile
index a3de2f8..7a614f9 100644
--- a/tools/perf/config/Makefile
+++ b/tools/perf/config/Makefile
@@ -106,6 +106,7 @@ FEATURE_TESTS =    \
  stackprotector-all  \
  stackprotector   \
  volatile-register-var  \
+ fortify-source   \
  libnuma
 
 $(foreach test,$(FEATURE_TESTS),$(call feature_check,$(test),$(test)))
@@ -123,7 +124,7 @@ ifeq ($(feature-volatile-register-var), 1)
 endif
 
 ifndef PERF_DEBUG
-  ifeq ($(call try-cc,$(SOURCE_HELLO),$(CFLAGS) -D_FORTIFY_SOURCE=2,-D_FORTIFY_SOURCE=2),y)
+  ifeq ($(feature-fortify-source), 1)
     CFLAGS += -D_FORTIFY_SOURCE=2
   endif
 endif
diff --git a/tools/perf/config/feature-checks/Makefile b/tools/perf/config/feature-checks/Makefile
index 5693299..529317e 100644
--- a/tools/perf/config/feature-checks/Makefile
+++ b/tools/perf/config/feature-checks/Makefile
@@ -4,6 +4,7 @@ FILES=     \
  test-stackprotector-all  \
  test-stackprotector  \
  test-volatile-register-var \
+ test-fortify-source  \
  test-libnuma
 
 CC := $(CC) -MD
@@ -26,6 +27,9 @@ test-stackprotector:
 test-volatile-register-var:
  $(BUILD) -Werror -Wvolatile-register-var
 
+test-fortify-source:
+ $(BUILD) -O2 -Werror -D_FORTIFY_SOURCE=2
+
 test-libnuma:
  $(BUILD) -lnuma
 
diff --git a/tools/perf/config/feature-checks/test-fortify-source.c b/tools/perf/config/feature-checks/test-fortify-source.c
new file mode 100644
index 0000000..c9f398d
--- /dev/null
+++ b/tools/perf/config/feature-checks/test-fortify-source.c
@@ -0,0 +1,6 @@
+#include <stdio.h>
+
+int main(void)
+{
+ return puts("hi");
+}
-- 
1.7.1